crates/bevy_pbr/src/render/pbr_lighting.wgsl

```diff
@@ -101,13 +101,17 @@ fn fresnel(f0: vec3<f32>, LoH: f32) -> vec3<f32> {
 
 // Cook-Torrance approximation of the microfacet model integration using Fresnel law F to model f_m
 // f_r(v,l) = { D(h,α) G(v,l,α) F(v,h,f0) } / { 4 (n⋅v) (n⋅l) }
-fn specular(f0: vec3<f32>, roughness: f32, h: vec3<f32>, NoV: f32, NoL: f32,
-              NoH: f32, LoH: f32, specularIntensity: f32) -> vec3<f32> {
+fn specular(f0: vec3<f32>, roughness: f32, h: vec3<f32>, NoV: f32, NoL: f32, NoH: f32, LoH: f32, specularIntensity: f32, f_ab: vec2<f32>) -> vec3<f32> {
     let D = D_GGX(roughness, NoH, h);
     let V = V_SmithGGXCorrelated(roughness, NoV, NoL);
     let F = fresnel(f0, LoH);
 
-    return (specularIntensity * D * V) * F;
+    var Fr = (specularIntensity * D * V) * F;
+
+    // Multiscattering approximation: https://google.github.io/filament/Filament.html#listing_energycompensationimpl
+    Fr *= 1.0 + f0 * (1.0 / f_ab.x - 1.0);
```

Comment:
the link does:
```hlsl
vec3 energyCompensation = 1.0 + f0 * (1.0 / dfg.y - 1.0);
```

I'm not sure what dfg here is in this context, I assume it's the same as `f_ab`, but it uses `.y` and you used `.x`. Was this intentional?

Reply:
It looks very wrong when swapped to `.y`. I forget why it's backwards vs filament, maybe filament has theirs backwards relative to unreal and others. I'm pretty sure it's correct, though.